xml_converter/src/xml_converter.cpp

```diff
@@ -168,27 +205,23 @@ void process_data(ParsedArguments parsed_arguments) {
         }
         cout << "Loading guildpoint pack " << marker_pack_config[i].path << endl;
 
-        map<string, vector<string>> top_level_category_file_locations = read_burrito_directory(
+        map<UniqueId, CategoryWithinSinglePack> top_level_category_file_locations = read_burrito_directory(
             marker_pack_config[i].path,
             &marker_categories,
             &parsed_pois
         );
-        for (map<string, vector<string>>::iterator it = top_level_category_file_locations.begin(); it != top_level_category_file_locations.end(); it++) {
-            top_level_category_file_locations_by_pack[it->first].push_back(it->second);
+        for (map<UniqueId, CategoryWithinSinglePack>::iterator it = top_level_category_file_locations.begin(); it != top_level_category_file_locations.end(); it++) {
+            top_level_category_file_locations_by_pack[it->first].categories.push_back(it->second);
         }
     }
     end = chrono::high_resolution_clock::now();
     dur = end - begin;
     ms = std::chrono::duration_cast<std::chrono::milliseconds>(dur).count();
     cout << "The guildpoint parse function took " << ms << " milliseconds to run" << endl;
 
-    for (map<string, vector<vector<string>>>::iterator it = top_level_category_file_locations_by_pack.begin(); it != top_level_category_file_locations_by_pack.end(); it++) {
-        if (it->second.size() != 1) {
-            for (size_t i = 0; i < it->second.size(); i++) {
-                for (size_t j = 0; j < it->second[i].size(); j++) {
-                    duplicate_categories[it->first].insert(it->second[i][j]);
-                }
-            }
+    for (map<UniqueId, CategoryWithinMultiplePacks>::iterator it = top_level_category_file_locations_by_pack.begin(); it != top_level_category_file_locations_by_pack.end(); it++) {
+        if (it->second.categories.size() != 1) {
```

Comment:
This should be `> 1` not `!= 1`. While not currently possible in the code, this `if` would trigger if `size()` were somehow `0` (maybe some sort of postprocessing filtering is done after the fact in the future). A `size_t` is unsigned and cant be negative so no need to worry about negative numbers.

If you are concerned about `0` it would be wise to add a sanity check `else if` to make sure that the `size()` is not 0, and have a unique error in that situation.